enhance: Update font size and height of drop down for mobile view

Tanuj1718 opened the pull request:
fixes: #34 
fixes #27


**Description**
1. Improve font size of navbar items for mobile view
2. Adjust the height of container which was overlapping with navbar items

https://github.com/user-attachments/assets/32f43a20-89be-4527-b801-52d507f4208a

rithviknishad commented on components/header.tsx:
```diff
@@ -252,13 +252,13 @@ export default function Header(props: { fixed?: boolean }) {
       />
       <nav
-        className={`flex relative items-stretch justify-between transition-all px-4 md:px-6 lg:px-8`}
+        className={`flex relative items-stretch justify-between transition-all px-4 md:px-6 lg:px-8 text-[5px]`}
```
Let's not use arbitrary values. Switch to using predefined sizes.

Tanuj1718 replied:
> Let's not use arbitrary values. Switch to using predefined sizes.

There are many other tags in which arbitrary values are used. Should I try to change them also or I need to change only mine which I have added?

rithviknishad commented on package-lock.json:
Discard changes made to the lockfile

Tanuj1718 commented:
@rithviknishad I have changed the files as you instructed. Can you please check it or is anything I'm missing?

shivankacker reviewed (changes requested):
<img width="208" alt="image" src="https://github.com/user-attachments/assets/ac6b5ccd-88fe-47aa-b3d1-cdf83f9c8cfe" />

Please reduce the distance between text

Tanuj1718 commented:
@shivankacker  Is it fine or do I need to adjust more? 

<img width="545" alt="Screenshot 2024-12-13 at 10 13 22 PM" src="https://github.com/user-attachments/assets/99630c57-0dee-4609-a70e-3ef3271b3dec" />

shivankacker commented:
@Tanuj1718 a little more would be good. Also we are adding "Timeline" to the header, so make sure it has enough space to accommodate for that too

Tanuj1718 commented:
@shivankacker  Done as you instructed. Also there is enough space for 'Timeline' tab in both mobile and web view.

<img width="353" alt="Screenshot 2024-12-17 at 8 49 10 PM" src="https://github.com/user-attachments/assets/cbbe5eb9-e43e-41c3-8c81-ba3e6c0cb9cb" />

shivankacker commented:
@Tanuj1718 lgtm, can you also fix the following bug with the PR:

clicking on an item on the menu is taking you to the corresponding page correctly, but should also close the menu.

Tanuj1718 commented:
> @Tanuj1718 lgtm, can you also fix the following bug with the PR:
> 
> clicking on an item on the menu is taking you to the corresponding page correctly, but should also close the menu.

Done @shivankacker

shivankacker reviewed (changes requested):
https://github.com/user-attachments/assets/9066bc71-6200-4e7d-a6b9-85bdb0e1f314

@Tanuj1718 doesn't seem to be fixed...

Tanuj1718 commented:
> Screen.Recording.2024-12-19.at.3.32.00.AM.mov 
> @Tanuj1718 doesn't seem to be fixed...

I have done it for supporters and contact tab. Is it not good for products and community tab? Because on hovering it , there is a dropdown menu appears and on clicking one of the options in drop down menu, the menu is getting closed.



https://github.com/user-attachments/assets/1e32ad85-9d1f-436e-a96e-ac6b8e5d5362

shivankacker commented:
@Tanuj1718 working for other options, but clicking on products > CARE does not close the menu. (Sometimes Next js will do a direct page change, where it seems to work, but when it changes the page dynamically, it doesnt. Keep clicking on navbar links to replicate)

Tanuj1718 commented:
> @Tanuj1718 working for other options, but clicking on products > CARE does not close the menu. (Sometimes Next js will do a direct page change, where it seems to work, but when it changes the page dynamically, it doesnt. Keep clicking on navbar links to replicate)

Done. Please let me know whether it is fine or not.

https://github.com/user-attachments/assets/9978c6bf-3d23-42f8-8fc3-4b0b4ec478a3

shivankacker commented on components/header.tsx:
```diff
@@ -82,7 +82,7 @@ export default function Header(props: { fixed?: boolean }) {
     { type: "dropdown", content: "Products", items: productsItems },
     { type: "dropdown", content: "Community", items: communityItems },
-    { type: "link", content: "Supporters", href: "/supporters" },
+    { type: "section", content: "Supporters", id: "supporters", page: "/supporters" },
```
Why this change? 🤔

Tanuj1718 replied:
> Why this change? 🤔

Because I want to use the section properties defined in 'switch' statements. Otherwise I have to write those properties in 'link' case.

shivankacker replied:
@Tanuj1718 a link must be used for links, you should find a way around this without disturbing existing elements

shivankacker commented on components/header.tsx:
```diff
@@ -159,6 +163,7 @@ export default function Header(props: { fixed?: boolean }) {
             href={item.page + "#" + item.id}
             className={className}
             onClick={(e) => {
+              setMobileMenuOpen(false);
```
This should come below the prevent default and stop propagation

shivankacker replied:
@Tanuj1718

shivankacker commented on components/header.tsx:
```diff
@@ -296,33 +308,37 @@ export default function Header(props: { fixed?: boolean }) {
-              }}
-            />
-          ))}
+          <div className={`${mobileMenuOpen ? "h-12" : "flex flex-row" }`}>
```
Changes here are possibly causing desktop dropdown to break


https://github.com/user-attachments/assets/07964ce5-b975-4e5a-915e-9f91574c9be6

Tanuj1718 replied:
> Changes here are possibly causing desktop dropdown to break
> 
>  Screen.Recording.2024-12-19.at.10.15.00.PM.mov

Yes, you are right. Thank you for pointing out.

shivankacker reviewed (changes requested):
@Tanuj1718 everything works well on mobile, but there are bugs on desktop now

shivankacker reviewed (changes requested):
![image](https://github.com/user-attachments/assets/4f1b4b5f-d7dd-4c1a-973e-e576f06b3d99)
- Did you increase the distance between titles? This seems to be more than it was before 🤔

Rest of the functionality looks good, but some code changes are required

Tanuj1718 commented:
> ![image](https://private-user-images.githubusercontent.com/23238460/398437716-4f1b4b5f-d7dd-4c1a-973e-e576f06b3d99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUwNjE4MzcsIm5iZiI6MTczNTA2MTUzNywicGF0aCI6Ii8yMzIzODQ2MC8zOTg0Mzc3MTYtNGYxYjRiNWYtZDdkZC00YzFhLTk3M2UtZTU3NmYwNmIzZDk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI0VDE3MzIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3ZDU3NmFmNjk1MzFmYzQyZTgxOGUwM2ZjODBmNzNhYjI2ZjBkODQ2MGUzZjU1NjJhMjBjMzdlNzJlZjNlM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ut61MslQpnp4gPDJD0lihioxWUFvLbJeYK3iNA3Hjz4)
> 
> * Did you increase the distance between titles? This seems to be more than it was before 🤔
> 
> Rest of the functionality looks good, but some code changes are required

Updated the code. Replaced the section with links for 'Support' item.

shivankacker commented:
@Tanuj1718 
<img width="362" alt="Screenshot 2024-12-31 at 3 40 27 PM" src="https://github.com/user-attachments/assets/290d681b-c9da-4bb5-a34f-4abc482f31ff" />

Tanuj1718 commented:
> @Tanuj1718 <img alt="Screenshot 2024-12-31 at 3 40 27 PM" width="362" src="https://private-user-images.githubusercontent.com/23238460/399451970-290d681b-c9da-4bb5-a34f-4abc482f31ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU2NDQ3MzksIm5iZiI6MTczNTY0NDQzOSwicGF0aCI6Ii8yMzIzODQ2MC8zOTk0NTE5NzAtMjkwZDY4MWItYzlkYS00YmI1LWEzNGYtNGFiYzQ4MmYzMWZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMxVDExMjcxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxZGI3NGE2NTIxYmFmZjJiYjc2Nzc5ZTJmZTNhMGE3ZTBkNmI0NjJlZDhlMzZiNGY2ZTA1ZGNjYjQzZDhkZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qbBqVexDDiO-UeLTqyZErqjHnNngLspsCVnox2H4YLg">

It is not showing when I am updating the code in my localhost,  but after restarting the terminal, it is showing these anonymous behaviour. Give me some time to find the exact problem and let me fix it then.

rithviknishad commented:
https://ubuntu.com/tutorials/install-ubuntu-desktop

Tanuj1718 commented:
> https://ubuntu.com/tutorials/install-ubuntu-desktop :)

Predefined className was creating issue. Now I have overwritten it by using ' !important '. Here's the updated reference: 


https://github.com/user-attachments/assets/171d0bbe-94fe-4af9-9942-e8f5227439bf

shivankacker commented:
Closing as `main` does not face this issue anymore. Thank you for contributing!
<img width="590" alt="Screenshot 2025-01-10 at 3 53 12 PM" src="https://github.com/user-attachments/assets/a50f52b4-7811-4126-ab67-51d97a5d3969" />
